machines/vic/reification/appliance.go

```diff
@@ -0,0 +1,416 @@
+	if err != nil {
+		err = errors.Errorf("Failed to get bridge network: %s", err)
+		return nil, err
+	}
+	d.networks["bridge"] = network
+	d.nics[conf.BridgeNetworkName] = "bridge"
+
+	// client network
+	network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ExtenalNetworkPath)
+	if err != nil {
+		err = errors.Errorf("Failed to get external network: %s", err)
+		return nil, err
+	}
+	d.networks["client"] = network
+	d.nics[conf.ExtenalNetworkName] = "client"
+
+	// management network
+	if conf.ManagementNetworkName != "" {
+		network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ManagementNetworkPath)
+		if err != nil {
+			err = errors.Errorf("Failed to get management network: %s", err)
+			return nil, err
+		}
+		d.networks["management"] = network
+		d.nics[conf.ManagementNetworkName] = "management"
+	}
+
+	var backing types.BaseVirtualDeviceBackingInfo
+	for _, network := range d.networks {
+		backing, err = network.EthernetCardBackingInfo(d.ctx)
+		if err != nil {
+			err = errors.Errorf("Failed to get network backing info for %s: %s", network, err)
+			return nil, err
+		}
+
+		net := &types.VirtualDeviceConfigSpec{
+			Operation: types.VirtualDeviceConfigSpecOperationAdd,
+			Device: &types.VirtualVmxnet3{
+				VirtualVmxnet: types.VirtualVmxnet{
+					VirtualEthernetCard: types.VirtualEthernetCard{
+						VirtualDevice: types.VirtualDevice{
+							Key:     -3,
+							Backing: backing,
+						},
+						AddressType: string(types.VirtualEthernetCardMacTypeGenerated),
+					},
+				},
+			},
+		}
+		devices = append(devices, net)
+	}
+	return devices, nil
+}
+
+func (d *Dispatcher) createApplianceSpec(conf *config.Config) (*types.VirtualMachineConfigSpec, error) {
+	var unitNumber int32
+	unitNumber = -1
+	thin := true
+	netDevices, err := d.getNetworkDevices(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	spec := &types.VirtualMachineConfigSpec{
+		Name:     conf.DisplayName,
+		GuestId:  "other3xLinux64Guest",
+		Files:    &types.VirtualMachineFileInfo{VmPathName: fmt.Sprintf("[%s]", conf.ImageDatastoreName)},
+		NumCPUs:  conf.NumCPUs,
+		MemoryMB: conf.MemoryMB,
+		ExtraConfig: []types.BaseOptionValue{
+			&types.OptionValue{Key: "guestinfo.vch/components", Value: "/sbin/docker-engine-server /sbin/port-layer-server /sbin/vicadmin"},
+			&types.OptionValue{Key: "guestinfo.vch/sbin/imagec", Value: "-debug -logfile=/var/log/vic/imagec.log -insecure"},
+			&types.OptionValue{Key: "guestinfo.vch/sbin/port-layer-server",
+				Value: fmt.Sprintf("--host=localhost --port=8080 --insecure --sdk=%s --datacenter=%s --cluster=%s --datastore=%s --network=%s --vch=%s",
+					conf.TargetPath, conf.DatacenterName, conf.ClusterPath, conf.ImageStorePath, conf.ExtenalNetworkPath, conf.DisplayName)},
+		},
+		DeviceChange: []types.BaseVirtualDeviceConfigSpec{
+			&types.VirtualDeviceConfigSpec{
+				FileOperation: types.VirtualDeviceConfigSpecFileOperationCreate,
```

Comment:
So I'm not clear about this. We don't need disk for VCH appliance, even in the future?